lab_6_pipeline/pipeline.py

```diff
@@ -27,16 +49,46 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = list(self.path_to_raw_txt_data.glob("*_raw.txt"))
+        meta_files = list(self.path_to_raw_txt_data.glob("*_meta.json"))
+
+        if len(raw_files) != len(meta_files):
+            raise InconsistentDatasetError
+
+        sorted_raw_files = sorted(raw_files, key=get_article_id_from_filepath)
+        sorted_meta_files = sorted(meta_files, key=get_article_id_from_filepath)
+
+        for ind, (raw, meta) in enumerate(zip(sorted_raw_files, sorted_meta_files)):
+            if ind + 1 != get_article_id_from_filepath(raw) or ind + 1 != \
+                    get_article_id_from_filepath(meta) or any(
+                    file.stat().st_size == 0 for file in (raw_files + meta_files)):
+                raise InconsistentDatasetError
 
     def _scan_dataset(self) -> None:
         """
         Register each dataset entry.
         """
+        self._storage = {get_article_id_from_filepath(file): from_raw(file, Article(
```

Comment:
could be better formatting:


```py

a = {
  key: value
  for ...
}
``` 

OR if long value expression

```py

a = {
  key: 
  value
  for ...
}
```

Reply:
changed it